nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -428,33 +428,45 @@ WaypointFollower::convertGPSPoses2MapPoses(
       }
       continue;
     } else {
-      // this poses are assumed to be on global frame (map)
-      geometry_msgs::msg::PoseStamped curr_pose_map_frame;
-      curr_pose_map_frame.header.frame_id = "map";
-      curr_pose_map_frame.header.stamp = stamp;
-      curr_pose_map_frame.pose.position.x = response->map_point.x;
-      curr_pose_map_frame.pose.position.y = response->map_point.y;
-      curr_pose_map_frame.pose.position.z = response->map_point.z;
-
-      geometry_msgs::msg::PoseStamped curr_pose_utm_frame;
-      curr_pose_utm_frame.pose.orientation = curr_oriented_navsat_fix.orientation;
-      curr_pose_utm_frame.header.frame_id = "utm";
+      // fromll_client converted the point from LL to Map frames
+      // but it actually did not consider the possible yaw offset between utm and map frames ;
+      // "https://github.com/cra-ros-pkg/robot_localization/blob/
+      // 79162b2ac53a112c51d23859c499e8438cf9938e/src/navsat_transform.cpp#L394"
+      // see above link on how they set rotation between UTM and
+      // Map to Identity , where actually it might not be
+      geometry_msgs::msg::TransformStamped utm_to_map_transform;
       try {
-        tf_buffer_->transform(
-          curr_pose_utm_frame, curr_pose_map_frame, "map",
-          tf2::durationFromSec(transform_tolerance_));
+        utm_to_map_transform = tf_buffer_->lookupTransform("utm", "map", tf2::TimePointZero);
```

Comment:
I don't see how these differ. In the previous version you give it the info and let it transform, now you're handling the same operations manually - both between the same 2 frames.